### PR TITLE
feat: Refactors `Button` component

### PR DESCRIPTION
**Pull request by lucasfp13**
## What's the purpose of this pull request?

This PR intends to refactor the `Button` and all its related components, including the use of CSS modules. Also, adds `Button` component's docs following the improvement on Storybook documentation defined here: [Unify Storybook Documentation #26](https://github.com/vtex-sites/gatsby.store/pull/26)

## How to test it?

The component behavior and appearance should be the same as before.

Changes can be seen in Storybook (`yarn storybook --docs`)

## Checklist

<em>You may erase this after checking them all ;)</em>

- [x] Added an entry in the `CHANGELOG.md` at the beginning of its due section. [The latest version should comes first](https://keepachangelog.com/en/1.0.0/#:~:text=The%20latest%20version%20comes%20first.).
- [x] Added the PR number with the PR link at the entry in the `CHANGELOG.md`. E.g., *New items in the `pull_request_template.md` ([#12](https://github.com/vtex-sites/gatsby.store/pull/12))* 

- PR description
- [x] Updated the Storybook - *if applicable*.
- [x] Added a label according to the PR goal - `Breaking change`, `Enhancement`, `Bug` or `Chore`.
- [x] Added the component, hook, or pathname in-between backticks (``) *- If applicable*. E.g., *`ComponentName` component*.

### CODE REVIEW

**Review by ArthurTriis1 — APPROVED** [review]

File: src/components/ui/Button/Button.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+  ) : (
+    <Button {...args}>Button</Button>
+  )
+}
+
+<header>
+
+# Button
+
+A element that allows users to take actions, make choices or trigger an event with a single tap.
```
Comment (hellofanny) [inline src/components/ui/Button/Button.stories.mdx:35]:
```suggestion
This component allows users to take actions, make choices, or trigger an event with a single tap.
```

File: src/components/ui/Button/Button.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+| <code>--fs-button-shadow-hover</code>        | <code>none</code>                                                                                 |
+|                                              |                                                                                                   |
+| <code>--fs-button-border-radius</code>       | <code>var(--fs-border-radius-default)</code>                                                      |
+| <code>--fs-button-border-width</code>        | <code>var(--fs-border-width-thick)</code>                                                         |
+| <code>--fs-button-border-color</code>        | <code>transparent</code>                                                                          |
+|                                              |                                                                                                   |
+| <code>--fs-button-text-size</code>           | <code>var(--fs-text-size-base)</code>                                                             |
+| <code>--fs-button-text-weight</code>         | <code>bold</code>                                                                                 |
```
Comment (hellofanny) [inline src/components/ui/Button/Button.stories.mdx:76]:
```suggestion
| <code>--fs-button-text-weight</code>         | <code>var(--fs-text-weight-bold)</code>                                                                                 |
```

Comment (hellofanny) [reply src/components/ui/Button/Button.stories.mdx:76]: Can you also update here pls? Thanks.


https://github.com/vtex-sites/gatsby.store/blob/d2eab5cf901a9969332c97565b091774ae2d1912/src/components/ui/Button/button.scss#L21

File: src/components/ui/Button/Button.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+| <code>--fs-button-padding</code>             | <code>calc(var(--fs-spacing-1) - (var(--fs-button-border-width) \* 2)) var(--fs-spacing-3)</code> |
+| <code>--fs-button-height</code>              | <code>var(--fs-control-tap-size)</code>                                                           |
+| <code>--fs-button-gap</code>                 | <code>var(--fs-spacing-2)</code>                                                                  |
+|                                              |                                                                                                   |
+| <code>--fs-button-shadow</code>              | <code>none</code>                                                                                 |
+| <code>--fs-button-shadow-hover</code>        | <code>none</code>                                                                                 |
+|                                              |                                                                                                   |
+| <code>--fs-button-border-radius</code>       | <code>var(--fs-border-radius-default)</code>                                                      |
```
Comment (hellofanny) [inline src/components/ui/Button/Button.stories.mdx:71]:
```suggestion
| <code>--fs-button-border-radius</code>       | <code>var(--fs-border-radius)</code>                                                      |
```

**Review by hellofanny — CHANGES_REQUESTED** [review]
Niceee! I've left a few suggestions, can you please take a look?

Shouldn't we also list `disabled` as a variation?

image

File: src/components/ui/Button/button.module.scss [hunk 4]
```diff
@@ -1,121 +1,121 @@
 @import "src/styles/scaffold";
 
-[data-fs-button] {
+.fs-button[data-fs-button] {
```
Comment (renatamottam) [inline src/components/ui/Button/button.module.scss:3]: No need for the `data-attribute` here! See `product-card.module.scss` as reference.
```suggestion
.fs-button {
```

**Review by renatamottam — APPROVED** [review]
Amazing work! Just left a fix to follow our pattern 🙃

File: src/components/ui/Button/Button.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,911 @@
...
+This component allows users to take actions, make choices or trigger an event with a single tap.
+
+</header>
+
+- [Overview](#overview)
+- [Usage](#usage)
+- [Nested Elements](#nested-elements)
+- [Hierarchy](#hierarchy)
+- [Variants](#variants)
+- [Related Components](#related-components)
```
Comment (eduardoformiga) [inline src/components/ui/Button/Button.stories.mdx:29]: Nice! 

@renatamottam should we add this section from now on?

**Review by eduardoformiga — APPROVED** [review]
Great Job! 👏 

LGTM after applying @renatamottam  suggestions.

File: src/components/ui/Button/Button.tsx [hunk 6]
```diff
@@ -1,36 +1,58 @@
...
+  /**
+   * Icon component
+   */
   icon?: ReactNode
+  /**
+   * Specifies where the icon should be positioned
+   */
   iconPosition?: IconPosition
+  /**
+   * For accessibility purposes, defines an ARIA label to the element when no label is provided
```
Comment (hellofanny) [inline src/components/ui/Button/Button.tsx:33]:
```suggestion
   * For accessibility purposes, defines an ARIA label to the button when no label is provided
```

**Review by hellofanny — APPROVED** [review]
Well done @lucasfp13 !!! 🚀🚀🚀 
I liked the `Related Components` section idea :) 

What do you think of adding a sample of usage of Button only with an icon (without label)? We could add an example with an aria-label for this case.

image

**Pull request merged**